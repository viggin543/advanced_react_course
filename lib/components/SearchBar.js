import React from 'react';
//'lodash.debounce' this way we import only the debounce function 
import debounce from 'lodash.debounce';
import StoreProvider from './StoreProvider';

//https://facebook.github.io/react/docs/refs-and-the-dom.html
{/* <input
type="text"
ref={(input) => { this.textInput = input; }} /> */}
class SearchBar extends React.Component {

  state = {
    searchTerm: ''
  }

  doSearch = debounce(() => {    
    this.props.store.search(this.state.searchTerm);
  }, 300);

  handleChange = (event) => {
    this.setState({ searchTerm: event.currentTarget.value },
      this.doSearch);
  }

  render() {
    return (
      <input
        type="search"
        placeholder="enter filter"
        onChange={this.handleChange}
        value={this.state.searchTerm}
      />
    );
  }
}
export default StoreProvider()(SearchBar);