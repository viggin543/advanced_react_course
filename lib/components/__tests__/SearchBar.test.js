import React from 'react';
import SearchBar from '../SearchBar';

// import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';

describe('SearchBar', () => {

  it('renders correctly', () => {
    const unit = shallow(<SearchBar/>);
    expect(unit).toMatchSnapshot();
  });
});