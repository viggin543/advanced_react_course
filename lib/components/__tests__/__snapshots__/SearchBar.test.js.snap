// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <SearchBar
    store={undefined}
  />,
  "nodes": Array [
    <SearchBar
      store={undefined}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <SearchBarContainer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Wrapper {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
        },
        "props": Object {},
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "usedState": [Function],
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <SearchBar
          store={undefined}
        />,
        "_debugID": 2,
        "_renderedOutput": <SearchBar
          store={undefined}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBarContainer />,
}
`;
